IonDotnet.Tests/Builders/LoaderTest.cs

```diff
@@ -21,23 +22,32 @@ public void CascadingSymtab_TestCorrectSymbolDecoding()
 
             Assert.AreEqual(3, datagram.Count);
 
-            //the first value $13 should be unknown text
-            Assert.AreEqual(IonType.Symbol, datagram[0].Type);
-            var token = ((IonSymbol) datagram[0]).SymbolValue;
-            Assert.AreEqual(13, token.Sid);
-            Assert.IsNull(token.Text);
-
-            //2nd value $10 should be rock:10
-            Assert.AreEqual(IonType.Symbol, datagram[1].Type);
-            token = ((IonSymbol) datagram[1]).SymbolValue;
-            Assert.AreEqual(10, token.Sid);
-            Assert.AreEqual("rock", token.Text);
+            int counter = 0;
+            foreach (var ionValue in datagram)
+            {
+                CascadingSymtabAssertion(ionValue, counter++);
+            }
+        }
 
-            //3rd value $10 should be unknown text
-            Assert.AreEqual(IonType.Symbol, datagram[0].Type);
-            token = ((IonSymbol) datagram[2]).SymbolValue;
-            Assert.AreEqual(10, token.Sid);
-            Assert.IsNull(token.Text);
+        private void CascadingSymtabAssertion(IIonValue ionValue, int itemNumber)
+        {
+            Assert.AreEqual(IonType.Symbol, ionValue.Type());
+            var token = ((IonSymbol)ionValue).SymbolValue;
```

Comment:
This cast is no longer necessary, correct?

Reply:
Correct. Done.